Part_1/Calculator/app/src/main/java/com/ensf409/calculator/Calculator.java

```diff
@@ -0,0 +1,223 @@
+		while(!numbers.isEmpty()){//while there are still numbers to pop, keep poppin'
+			char op = operations.pop();
+			Double num = numbers.pop();
+			if (op == '+')
+				sum += num;
+			else if (op == '-')
+				sum -= num;
+			else if (op == '*')
+				sum *= num;
+			else if (op == '/'){
+				//if a divide by zero is attempted, make note of it
+				if(num == 0)
+					divZeroErr = true;
+				sum /= num;
+			}
+			//should never reach this else statement if GUI is properly implemented
+			else
+				System.out.println("Operation not recognized...");
+		}
+		lastSum = sum;
+	}
+	//first version of overloaded ButtonPressed, accepts integer to build up currentNumber
+	public String ButtonPressed(int num){
+		//if divZeroErr has occurred, "lock out" method until 'C' has been pressed
+		if(divZeroErr)
+			return "Divide Zero Error";
+		double asDouble = (double)num;
+		//if decimal has been used on current number
+		if(afterDec!= 0){
+			//compute arithmetic for currentNumber
+			currentNumber += asDouble*Math.pow(10, (afterDec));
+			//this code is to produce a proper string toDisplay just in case 0's have been pressed after decimal place
+			int places = -1;
+			places *= (int)afterDec;
+			DecimalFormat df = new DecimalFormat("#.##");
+			df.setMaximumFractionDigits(places);
+			df.setMinimumFractionDigits(places);
+			toDisplay = df.format(currentNumber);
+			//decrement afterDec for proper arithmetic on next call of ButtonPressed
+			afterDec--;
+		}
+		//if decimal has not been used on currentNumber
+		else{
+			//compute arithmetic for currentNumber
+			currentNumber = (currentNumber*10) + asDouble;
+			toDisplay = Integer.toString((int)currentNumber.doubleValue());
+		}
+		//set lastReceived to 'N' notifying a number was entered last
+		lastReceived = 'N';
+		return toDisplay;
+	}
+	//first version of overloaded ButtonPressed, accepts a char, 4 possible char types 'O'(operation), 'C'(cancel), '=', or '.'
+	public String ButtonPressed(char ch){
+		//if '=' pressed
+		if(ch == '='){
+			//if divZeroErr has occurred, "lock out" method until 'C' has been pressed
+			if(divZeroErr)
+				return "Divide Zero Error";
+			//if a operation was just pressed, display a format error to screen
+			else if(lastReceived == 'O'){
+				return "Format ERROR";
+			}
+			//if '=' was just pressed, display lastSum again, if 'C' was just pressed, display blank again
+			else if(lastReceived == '=' || lastReceived == 'C'){
+				return toDisplay;
+			}
+			//if '.' or 'N' was pressed last
+			else{
+				//add currentNumber to the list and calculate the sum
+				numbers.add(currentNumber);
+				calcSum();
+				//if a divide by zero occured during calcSum, display error, do not pass Go, do not collect $200
+				if(divZeroErr)
+					return "Divide Zero Error";
+				//display the sum that was calculated, clear, and set lastReceived to '='
+				toDisplay = Double.toString(lastSum);
+				clear();
+				lastReceived = ch;
+				return toDisplay;
```

Comment:
It looks like all the if statements return toDisplay, so should we just have it once at the end?